doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,281 @@
+advanced configuration functionality.
+Not all the users will need them, if the default configuration of
+the Ostro OS meets their needs.
+
+The process is exemplified by using real components and describing
+what each step looks like, when applied to such components.
+Unless stated otherwise, the changes described are to be intended
+for the file ``conf/local.conf``
+
+
+Step 0: Review and optionally modify the content of the os-core bundle
+----------------------------------------------------------------------
+The os-core bundle comes with a predefined set of components,
+however each device is likely to have slighlty different requirements.
+In some cases it might be necessary to adjust the set of components
+included in os-core.
+
+To do this, either modify or extend the ``OSTRO_IMAGE_FEATURES_REFERENCE``
+and the ``OSTRO_IMAGE_INSTALL_REFERENCE`` variables for changes
+respectively to the feature and component sets.
+
+Why modifying os-core? Because it's the only unremovable bundle and
+therefore it is well suited to deliver to the device all those tools
+that must always be available.
+
+Example::
+
+  ``OSTRO_IMAGE_INSTALL_REFERENCE += "sudo"``
+
+will add the content of the ``sudo`` recipe to the bundle os-core.
+
+.. note::
+   The step shown works only for adding components to os-core.
+
+   This is the typical case when customizing Ostro OS, because the
+   os-core configuration is very minimal.
+
+   However, in case one needs to remove something from os-core,
+   it is necessary to create either a new image recipe or a new
+   .bbappend file, where to modify the variables that define the
+   content of os-core.
+
+   These are the variable that must be modified::
+
+   - OSTRO_IMAGE_FEATURES_REFERENCE
+   - OSTRO_IMAGE_INSTALL_REFERENCE
+   - IMAGE_INSTALL
+
+   The recommended way is to modify a copy of the original recipe
+   ``ostro-image.bb`` and refer to ``ostro-image.bbclass`` for
+   the values to use.
+
+
+Step 1: Review and optionally modify the list of bundles
+--------------------------------------------------------
+Ostro already defines some bundles, and users can define additional ones.
+
+Example::
+
+  SWUPD_BUNDLES += "sudo_feature"
+  BUNDLE_CONTENTS[sudo_feature] = "sudo"
+
+This example is an alternative to add the component that was shown in
+Step 0.  Instead of extending the content of os-core, put the extra
+component in its own bundle. The main difference is that this
+approach will not install the component by default in any image.
+
+
+Step 2: Review and optionally modify the content of the images created
+----------------------------------------------------------------------
+It is also possible to choose which bundles will be pre-installed on the
+customized image, by:
+
+#. adding the bundle to the list of images generated for bundle purposes
+
+   Example::
+
+     SWUPD_IMAGES += "\
+       pre_installed_content \
+     "
+
+#. adding a varflag to SWUPD_IMAGES.
```

Comment:
You wanted to change "adding a varflag to SWUPD_IMAGES." didn't you?

From the earlier comment: Instead of explaining the syntax, better mention what this does, as in "Define the content of the "pre_installed_content" image by setting the corresponding variable flag in SWUPD_IMAGES:"